proto/destination.proto

```diff
@@ -103,12 +103,16 @@ message WeightedAddr {
   AuthorityOverride authority_override = 7;
 }
 
-// Which strategy should be used for verifying TLS.
 message TlsIdentity {
   reserved 2;
   reserved "k8s_pod_identity";
 
-  oneof strategy { DnsLikeIdentity dns_like_identity = 1; }
+  oneof strategy {
+    DnsLikeIdentity dns_like_identity = 1;
+    UriLikeIdentity uri_like_identity = 3;
+  }
+
+  ServerName server_name = 4;
```

Comment:
It seems appropriate to use DnsLikeIdentity for the server_name type to me.  The server name has to be a DNS-like string.